src/main/java/appeng/services/export/MinecraftItemCSVExporter.java

```diff
@@ -0,0 +1,261 @@
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.google.common.base.Function;
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+
+import org.apache.commons.io.FileUtils;
+
+import net.minecraft.block.Block;
+import net.minecraft.creativetab.CreativeTabs;
+import net.minecraft.init.Blocks;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.StatCollector;
+
+import cpw.mods.fml.common.registry.FMLControlledNamespacedRegistry;
+
+import appeng.core.AELog;
+
+
+/**
+ * handles the exporting including processing, transformation and persisting the information
+ *
+ * @author thatsIch
+ * @version rv3 - 14.08.2015
+ * @since rv3 14.08.2015
+ */
+final class MinecraftItemCSVExporter implements Exporter
+{
+	@Nonnull
+	private final File exportDirectory;
+	@Nonnull
+	private final FMLControlledNamespacedRegistry<Item> itemRegistry;
+	@Nonnull
+	private final ExportMode mode;
+
+	/**
+	 * @param exportDirectory directory of the resulting export file. Non-null required.
+	 * @param itemRegistry the registry with minecraft items. Needs to be populated at that time, thus the exporting can only happen in init (pre-init is the
+	 * phase when all items are determined)
+	 * @param mode mode in which the export should be operated. Resulting CSV will change depending on this.
+	 */
+	public MinecraftItemCSVExporter( @Nonnull File exportDirectory, @Nonnull FMLControlledNamespacedRegistry<Item> itemRegistry, @Nonnull ExportMode mode )
+	{
+		this.exportDirectory = Preconditions.checkNotNull( exportDirectory );
+		Preconditions.checkArgument( !exportDirectory.isFile() );
+		this.itemRegistry = Preconditions.checkNotNull( itemRegistry );
+		this.mode = Preconditions.checkNotNull( mode );
+	}
+
+	@Override
+	public void export()
+	{
+		final Iterable<Item> items = this.itemRegistry.typeSafeIterable();
+		final List<Item> itemList = Lists.newArrayList( items );
+
+		final List<String> lines = Lists.transform( itemList, new ItemRowExtractFunction( this.itemRegistry, this.mode ) );
+
+		final Joiner lineJoiner = Joiner.on( '\n' ).skipNulls();
+		final String joined = lineJoiner.join( lines );
+
+		final String fileName = "items.csv";
+		final File file = new File( this.exportDirectory, fileName );
+
+		try
+		{
+			FileUtils.forceMkdir( this.exportDirectory );
+
+			final FileWriter writer = new FileWriter( file );
+			writer.write( "Mod:Item:MetaData, Localized Name" );
+			if( this.mode == ExportMode.VERBOSE )
+			{
+				writer.write( ", Unlocalized Name, Is Air?, Class Name" );
+			}
+			writer.write( "\n" );
+
+			writer.write( joined );
+			writer.flush();
+			writer.close();
+
+			AELog.info( "Exported successfully %d items into %s", lines.size(), fileName );
+		}
+		catch( IOException e )
+		{
+			e.printStackTrace();
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1148)